WordPress/Classes/ViewRelated/Reader/ReaderDetailCoordinator.swift

```diff
@@ -156,6 +165,23 @@ class ReaderDetailCoordinator {
         viewController?.navigationController?.popViewController(animated: true)
     }
 
+    /// Some posts have content from private sites that need special cookies
+    ///
+    /// Use this method to make sure these cookies are downloaded.
+    /// - Parameter webView: the webView where the post will be rendered
+    /// - Parameter completion: a completion block
+    func authenticateIfNeeded(in webView: WKWebView, completion: @escaping () -> Void) {
+        guard let postURLString = post?.permaLink,
+            let postURL = URL(string: postURLString) else {
+            completion()
+            return
```

Comment:
Same question as above... is this an acceptable / normal / regular scenario?  Otherwise consider loggin an error here.

Reply:
I'd say yes. If the permalink doesn't exist the post will be rendered either way.